metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/utils/RdfEntityUtils.java

```diff
@@ -0,0 +1,158 @@
+package eu.europeana.enrichment.utils;
+
+import eu.europeana.enrichment.api.internal.AggregationFieldType;
+import eu.europeana.enrichment.api.internal.ProxyFieldType;
+import eu.europeana.enrichment.api.internal.SearchTerm;
+import eu.europeana.enrichment.api.internal.SearchTermContext;
+import eu.europeana.metis.schema.jibx.AboutType;
+import eu.europeana.metis.schema.jibx.Aggregation;
+import eu.europeana.metis.schema.jibx.EuropeanaType;
+import eu.europeana.metis.schema.jibx.ProxyType;
+import eu.europeana.metis.schema.jibx.RDF;
+import eu.europeana.metis.schema.jibx.ResourceOrLiteralType;
+import eu.europeana.metis.schema.jibx.ResourceOrLiteralType.Lang;
+import eu.europeana.metis.schema.jibx.ResourceOrLiteralType.Resource;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Utility class for {@link ProxyType} operations in the {@link RDF}
+ */
+public final class RdfEntityUtils {
+
+  private RdfEntityUtils() {
+  }
+
+  /**
+   * Add a link to the specified {@link AboutType} to the EuropeanaProxy.
+   *
+   * @param rdf the rdf to append to
+   * @param link the about value to link
+   * @param linkTypes the types of the link to add in the europeana proxy.
+   */
+  public static void appendLinkToEuropeanaProxy(RDF rdf, String link, Set<ProxyFieldType> linkTypes) {
+    final Map<ProxyFieldType, Set<String>> allProxyLinksPerType = getAllProxyLinksPerType(rdf);
+    final ProxyType europeanaProxy = getEuropeanaProxy(rdf);
+    for (ProxyFieldType linkType : linkTypes) {
+      final boolean alreadyExists = Optional.ofNullable(allProxyLinksPerType.get(linkType))
+          .orElseGet(Collections::emptySet).contains(link);
+      if (!alreadyExists) {
+        final List<EuropeanaType.Choice> choices = Optional
+            .ofNullable(europeanaProxy.getChoiceList()).orElseGet(ArrayList::new);
+        choices.add(linkType.createChoice(link));
+        europeanaProxy.setChoiceList(choices);
+      }
+    }
+    replaceProxy(rdf, europeanaProxy);
+  }
+
+  /**
+   * Replace matching aggregation values with their found corresponding links.
+   *
+   * @param rdf the rdf to update
+   * @param link the about value to use
+   * @param searchTermAggregation the aggregation search term to use for finding the matched values
+   */
+  public static void replaceValueWithLinkInAggregation(RDF rdf, String link,
+      SearchTermContext searchTermAggregation) {
+    final List<Aggregation> aggregationList = rdf.getAggregationList();
+    for (AggregationFieldType aggregationFieldType : searchTermAggregation.getFieldTypes().stream()
+        .map(AggregationFieldType.class::cast).collect(Collectors.toList())) {
```

Comment:
No need to cast, right? The method called on these objects is available for the super type as well.

Reply:
Correct!